keps/sig-cli/5295-kyaml/README.md

```diff
@@ -0,0 +1,1500 @@
+For example, unlike standard YAML output,
+this dialect is not whitespace-sensitive,
+which makes it vastly easier to patch correctly in things like Helm charts.
+
+This KEP further proposes to make KYAML the standard format
+for all project-owned documentation and examples.
+
+## Motivation
+
+The vast majority of examples of how to use Kubernetes include YAML.
+These all use the most “conventional” YAML syntax,
+which is what `kubectl get -o yaml` also uses.
+Simple YAML is easy to read, but complex YAML can become very tricky.
+Some of the things that motivated this KEP are:
+
+Whitespace being significant means that
+readers and writers must track the nesting depth.
+While this does not sound too onerous,
+We have ample evidence that it is, in fact, very hard.
+Wrongly indented YAML is syntactically valid but does not marshal properly.
+Perhaps worst of all,
+template-expanding systems,
+such as the widely used `helm`,
+force users to get whitespace “just right”
+while **outside of** the main YAML context.
+
+String-quoting in YAML is generally optional,
+but some things that appear to be strings are in fact coerced to other types.
+For example, without quotes
+`NO`, `no`, `N`, `YES`, `yes`, `Y`, `On`, and `Off` are all treated as bool
+(aka  [“The Norway Bug”](https://hitchdev.com/strictyaml/why/implicit-typing-removed/)),
+`_42`, `_4_2_`, and `__________4__________2__________` are all treated as numbers,
+and `11:00` is also treated as a number (base 60).
+
+As an alternative to YAML most of Kubernetes also supports JSON,
+But JSON has its own drawbacks.
+It does not support comments.
+It is very particular about trailing commas.
+It requires quoted keys.
+
+In addition to `kubectl` output,
+most Kubernetes config files are either YAML or JSON.
+JSON lacks comments,
+which are very valuable for configuration,
+so most configuration and tooling is built around YAML.
+
+The most frustrating part of YAML is that it doesn’t have to be this bad.
+YAML is a very broad specification.
+Inside that specification is a reasonable grammar screaming to be let out.
+
+If we change our tooling and examples to use a less error-prone dialect of YAML,
+we can encourage the ecosystem to avoid some of the YAML pitfalls
+and reduce common mistakes and frustration.
+
+### Goals
+
+1. Specify a YAML dialect which is 100% compatible with existing parsers and tooling.
+2. Provide libraries and tooling for encoding and converting to this formatting.
+3. Provide a KYAML output format for kubectl
+4. Update examples and documentation throughout the project
+5. Address common YAML pitfalls through an opinionated formatting approach.
+6. Documenting the design and reasons for KYAML and the 100% compatibility with tooling expecting arbitrary YAML.
+
+### Non-Goals
+
+1. Introducing alternative configuration languages that are not compatible with existing tooling.
+2. Server-side support for KYAML or YAML
+3. Requiring projects to migrate by default.
```

Comment:
```suggestion
3. Require projects to migrate to by default.
```
Also which projects and migrate to what? I assume it's about non-k8s owned (non-k8s gh org)? Whereas the latter is about kyaml?

Reply:
We are not going to require kubeadm to migrate to kyaml, nor helm or any others.